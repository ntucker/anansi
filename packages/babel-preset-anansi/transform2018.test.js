const babel = require('@babel/core');

const buildPreset = require('./index');

describe('buildPreset - Babel Transform', () => {
  let api;

  beforeEach(() => {
    const {
      default: getTargets,
    } = require('@babel/helper-compilation-targets');
    api = {
      assertVersion: jest.fn(),
      env: jest.fn(),
      caller: jest.fn(cb => cb({ name: '@babel/cli' })),
      targets: jest.fn(() => getTargets()),
    };
    process.env.BROWSERSLIST_ENV = '2018';
    process.env.NODE_ENV = 'development';
  });

  const transformCode = (
    code,
    options = {
      hasJsxRuntime: true,
      loose: true,
    },
  ) => {
    const preset = buildPreset(api, options);
    return babel.transformSync(code, {
      filename: 'file.tsx',
      presets: [preset],
    }).code;
  };

  it('should transform JSX in development', () => {
    api.env.mockReturnValue('development');
    const code = `const element = <div>Hello World</div>;`;
    // remove first line (var _jsxFileName = ???)
    const transformedCode = transformCode(code).split('\n').slice(1).join('\n');
    expect(transformedCode).toMatchSnapshot();
  });

  it('should transform JSX in production', () => {
    api.env.mockReturnValue('production');
    const code = `const element = <div>Hello World</div>;`;
    const transformedCode = transformCode(code);
    expect(transformedCode).toMatchSnapshot();
  });

  it('should transform TypeScript to JavaScript', () => {
    api.env.mockReturnValue('development');
    const code = `const x: number = 42;`;
    const transformedCode = transformCode(code);
    expect(transformedCode).toMatchSnapshot();
  });

  it('should not apply class properties transform when loose', () => {
    api.env.mockReturnValue('development');
    const code = `class MyClass { declare myThing; myProp: number = 42; }`;
    const transformedCode = transformCode(code, {
      hasJsxRuntime: true,
      loose: true,
    });
    expect(transformedCode).toMatchSnapshot();
  });

  it('should apply class properties transform when not loose', () => {
    api.env.mockReturnValue('development');
    const code = `class MyClass { declare myThing; myProp: number = 42; }`;
    const transformedCode = transformCode(code, {
      hasJsxRuntime: true,
      loose: false,
    });
    expect(transformedCode).toMatchSnapshot();
  });

  it('should transform class statics', () => {
    api.targets.mockReturnValue('development');
    const code = `
    abstract class StaticEntity extends Entity {
      declare static a: string;
      static urlRoot = '/2/';

      static {
        this.a = this.urlRoot;
      }
    }
    `;
    const transformedCode = transformCode(code, {
      hasJsxRuntime: true,
      loose: true,
    });
    expect(transformedCode).toMatchSnapshot();
  });

  it('should transform Object.hasOwn() with usage-pure', () => {
    api.env.mockReturnValue('development');
    const code = `class MyClass { declare myThing; myProp: number = 42; }console.log(Object.hasOwn({ a: 1 }, 'a') ? 'yes' : 'no');`;
    const transformedCode = transformCode(code, {
      hasJsxRuntime: true,
      polyfillMethod: 'usage-pure',
    });
    expect(transformedCode).toMatchSnapshot();
  });

  it('should transform Object.hasOwn() with name==@babel/runtime', () => {
    api.env.mockReturnValue('development');
    api.caller.mockImplementation(cb => cb({ name: '@babel/cli' }));
    const code = `class MyClass { declare myThing; myProp: number = 42; }console.log(Object.hasOwn({ a: 1 }, 'a') ? 'yes' : 'no');`;
    const transformedCode = transformCode(code, {
      hasJsxRuntime: true,
    });
    expect(transformedCode).toMatchSnapshot();
  });

  it('should transform Object.hasOwn() with name==rollup-plugin-babel', () => {
    api.env.mockReturnValue('development');
    api.caller.mockImplementation(cb =>
      cb({
        name: 'rollup-plugin-babel',
        supportsStaticESM: true,
        supportsDynamicImport: true,
      }),
    );
    const code = `class MyClass { declare myThing; myProp: number = 42; }console.log(Object.hasOwn({ a: 1 }, 'a') ? 'yes' : 'no');`;
    const transformedCode = transformCode(code, {
      hasJsxRuntime: true,
    });
    expect(transformedCode).toMatchSnapshot();
  });

  it('should transform Object.hasOwn() with caller.library', () => {
    api.env.mockReturnValue('development');
    api.caller.mockImplementation(cb => cb({ library: true }));
    const code = `class MyClass { declare myThing; myProp: number = 42; }console.log(Object.hasOwn({ a: 1 }, 'a') ? 'yes' : 'no');`;
    const transformedCode = transformCode(code, {
      hasJsxRuntime: true,
    });
    expect(transformedCode).toMatchSnapshot();
  });

  it('should import Object.hasOwn() polyfill with usage-global', () => {
    api.env.mockReturnValue('development');
    const code = `class MyClass { declare myThing; myProp: number = 42; }console.log(Object.hasOwn({ a: 1 }, 'a') ? 'yes' : 'no');`;
    const transformedCode = transformCode(code, {
      hasJsxRuntime: true,
      polyfillMethod: 'usage-global',
    });
    expect(transformedCode).toMatchSnapshot();
  });

  it('should select polyfill entries with entry-global', () => {
    api.env.mockReturnValue('development');
    const code = `import 'core-js';class MyClass { declare myThing; myProp: number = 42; }`;
    const transformedCode = transformCode(code, {
      hasJsxRuntime: true,
      polyfillMethod: 'entry-global',
    });
    expect(transformedCode).toMatchSnapshot();
  });

  it('should not transform entry polyfill import when polyfillMethod: `false`', () => {
    api.env.mockReturnValue('development');
    const code = `import 'core-js';class MyClass { declare myThing; myProp: number = 42; }`;
    const transformedCode = transformCode(code, {
      hasJsxRuntime: true,
      polyfillMethod: false,
    });
    expect(transformedCode).toMatchSnapshot();
  });

  it('should not polyfill when polyfillMethod: `false`', () => {
    api.env.mockReturnValue('development');
    const code = `class MyClass { declare myThing; myProp: number = 42; }console.log(Object.hasOwn({ a: 1 }, 'a') ? 'yes' : 'no');`;
    const transformedCode = transformCode(code, {
      hasJsxRuntime: true,
      polyfillMethod: false,
    });
    expect(transformedCode).toMatchSnapshot();
  });

  it('should use core-js-pure by default when caller is rollup', () => {
    api.env.mockReturnValue('development');
    process.env.NODE_ENV = 'development';
    api.caller.mockImplementation(cb => cb({ name: '@rollup/plugin-babel' }));
    const code = `class MyClass { declare myThing; myProp: number = 42; }console.log(Object.hasOwn({ a: 1 }, 'a') ? 'yes' : 'no');`;
    let transformedCode = transformCode(code, { loose: false });
    expect(transformedCode).toContain('core-js-pure');
    expect(transformedCode).toContain('@babel/runtime-corejs3');
    api.caller.mockImplementation(cb => cb({ name: 'rollup-plugin-babel' }));
    transformedCode = transformCode(code, { loose: false });
    expect(transformedCode).toContain('core-js-pure');
    expect(transformedCode).toContain('@babel/runtime-corejs3');
  });

  it('should prefer pure core-js when it is explicitly listed in pkg deps', () => {
    api.env.mockReturnValue('development');
    process.env.NODE_ENV = 'development';
    api.caller.mockImplementation(cb => cb({ name: 'webpack' }));
    jest.mock(`${process.cwd()}/package.json`, () => ({
      dependencies: {
        '@babel/runtime-corejs3': '^7.26.0',
        'core-js-pure': '^3.40.0',
      },
    }));
    const code = `class MyClass { declare myThing; myProp: number = 42; }console.log(Object.hasOwn({ a: 1 }, 'a') ? 'yes' : 'no');`;
    let transformedCode = transformCode(code, { loose: false });
    expect(transformedCode).toContain('core-js-pure');
    expect(transformedCode).toContain('@babel/runtime-corejs3');
  });
});
