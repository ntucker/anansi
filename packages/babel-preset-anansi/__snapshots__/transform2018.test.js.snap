// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`buildPreset - Babel Transform should apply class properties transform when not loose 1`] = `
"import _defineProperty from "@babel/runtime-corejs3/helpers/esm/defineProperty";
class MyClass {
  constructor() {
    _defineProperty(this, "myProp", 42);
  }
}"
`;

exports[`buildPreset - Babel Transform should import Object.hasOwn() polyfill with usage-global 1`] = `
"import _defineProperty from "@babel/runtime-corejs3/helpers/esm/defineProperty";
import "core-js/modules/es.object.has-own.js";
class MyClass {
  constructor() {
    _defineProperty(this, "myProp", 42);
  }
}
console.log(Object.hasOwn({
  a: 1
}, 'a') ? 'yes' : 'no');"
`;

exports[`buildPreset - Babel Transform should not apply class properties transform when loose 1`] = `
"class MyClass {
  constructor() {
    this.myProp = 42;
  }
}"
`;

exports[`buildPreset - Babel Transform should not polyfill when polyfillMethod: \`false\` 1`] = `
"import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
class MyClass {
  constructor() {
    _defineProperty(this, "myProp", 42);
  }
}
console.log(Object.hasOwn({
  a: 1
}, 'a') ? 'yes' : 'no');"
`;

exports[`buildPreset - Babel Transform should not transform entry polyfill import when polyfillMethod: \`false\` 1`] = `
"import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import 'core-js';
class MyClass {
  constructor() {
    _defineProperty(this, "myProp", 42);
  }
}"
`;

exports[`buildPreset - Babel Transform should select polyfill entries with entry-global 1`] = `
"import _defineProperty from "@babel/runtime-corejs3/helpers/esm/defineProperty";
import "core-js/modules/es.symbol.description.js";
import "core-js/modules/es.symbol.async-dispose.js";
import "core-js/modules/es.symbol.async-iterator.js";
import "core-js/modules/es.symbol.dispose.js";
import "core-js/modules/es.symbol.match.js";
import "core-js/modules/es.symbol.match-all.js";
import "core-js/modules/es.symbol.replace.js";
import "core-js/modules/es.symbol.search.js";
import "core-js/modules/es.symbol.split.js";
import "core-js/modules/es.error.cause.js";
import "core-js/modules/es.error.is-error.js";
import "core-js/modules/es.aggregate-error.js";
import "core-js/modules/es.aggregate-error.cause.js";
import "core-js/modules/es.suppressed-error.constructor.js";
import "core-js/modules/es.array.at.js";
import "core-js/modules/es.array.find-last.js";
import "core-js/modules/es.array.find-last-index.js";
import "core-js/modules/es.array.flat.js";
import "core-js/modules/es.array.flat-map.js";
import "core-js/modules/es.array.includes.js";
import "core-js/modules/es.array.iterator.js";
import "core-js/modules/es.array.reduce.js";
import "core-js/modules/es.array.reduce-right.js";
import "core-js/modules/es.array.reverse.js";
import "core-js/modules/es.array.sort.js";
import "core-js/modules/es.array.to-reversed.js";
import "core-js/modules/es.array.to-sorted.js";
import "core-js/modules/es.array.to-spliced.js";
import "core-js/modules/es.array.unscopables.flat.js";
import "core-js/modules/es.array.unscopables.flat-map.js";
import "core-js/modules/es.array.with.js";
import "core-js/modules/es.array-buffer.constructor.js";
import "core-js/modules/es.array-buffer.slice.js";
import "core-js/modules/es.data-view.get-float16.js";
import "core-js/modules/es.data-view.set-float16.js";
import "core-js/modules/es.array-buffer.detached.js";
import "core-js/modules/es.array-buffer.transfer.js";
import "core-js/modules/es.array-buffer.transfer-to-fixed-length.js";
import "core-js/modules/es.disposable-stack.constructor.js";
import "core-js/modules/es.global-this.js";
import "core-js/modules/es.iterator.constructor.js";
import "core-js/modules/es.iterator.dispose.js";
import "core-js/modules/es.iterator.drop.js";
import "core-js/modules/es.iterator.every.js";
import "core-js/modules/es.iterator.filter.js";
import "core-js/modules/es.iterator.find.js";
import "core-js/modules/es.iterator.flat-map.js";
import "core-js/modules/es.iterator.for-each.js";
import "core-js/modules/es.iterator.from.js";
import "core-js/modules/es.iterator.map.js";
import "core-js/modules/es.iterator.reduce.js";
import "core-js/modules/es.iterator.some.js";
import "core-js/modules/es.iterator.take.js";
import "core-js/modules/es.iterator.to-array.js";
import "core-js/modules/es.json.stringify.js";
import "core-js/modules/es.map.group-by.js";
import "core-js/modules/es.math.f16round.js";
import "core-js/modules/es.math.hypot.js";
import "core-js/modules/es.number.parse-float.js";
import "core-js/modules/es.number.parse-int.js";
import "core-js/modules/es.number.to-exponential.js";
import "core-js/modules/es.number.to-fixed.js";
import "core-js/modules/es.object.assign.js";
import "core-js/modules/es.object.from-entries.js";
import "core-js/modules/es.object.group-by.js";
import "core-js/modules/es.object.has-own.js";
import "core-js/modules/es.parse-float.js";
import "core-js/modules/es.parse-int.js";
import "core-js/modules/es.promise.js";
import "core-js/modules/es.promise.all-settled.js";
import "core-js/modules/es.promise.any.js";
import "core-js/modules/es.promise.finally.js";
import "core-js/modules/es.promise.try.js";
import "core-js/modules/es.promise.with-resolvers.js";
import "core-js/modules/es.array.from-async.js";
import "core-js/modules/es.async-disposable-stack.constructor.js";
import "core-js/modules/es.async-iterator.async-dispose.js";
import "core-js/modules/es.reflect.set.js";
import "core-js/modules/es.reflect.to-string-tag.js";
import "core-js/modules/es.regexp.constructor.js";
import "core-js/modules/es.regexp.escape.js";
import "core-js/modules/es.regexp.dot-all.js";
import "core-js/modules/es.regexp.exec.js";
import "core-js/modules/es.regexp.flags.js";
import "core-js/modules/es.regexp.test.js";
import "core-js/modules/es.regexp.to-string.js";
import "core-js/modules/es.set.difference.v2.js";
import "core-js/modules/es.set.intersection.v2.js";
import "core-js/modules/es.set.is-disjoint-from.v2.js";
import "core-js/modules/es.set.is-subset-of.v2.js";
import "core-js/modules/es.set.is-superset-of.v2.js";
import "core-js/modules/es.set.symmetric-difference.v2.js";
import "core-js/modules/es.set.union.v2.js";
import "core-js/modules/es.string.at-alternative.js";
import "core-js/modules/es.string.ends-with.js";
import "core-js/modules/es.string.includes.js";
import "core-js/modules/es.string.is-well-formed.js";
import "core-js/modules/es.string.match.js";
import "core-js/modules/es.string.match-all.js";
import "core-js/modules/es.string.replace.js";
import "core-js/modules/es.string.replace-all.js";
import "core-js/modules/es.string.search.js";
import "core-js/modules/es.string.split.js";
import "core-js/modules/es.string.starts-with.js";
import "core-js/modules/es.string.to-well-formed.js";
import "core-js/modules/es.string.trim.js";
import "core-js/modules/es.string.trim-end.js";
import "core-js/modules/es.string.trim-start.js";
import "core-js/modules/es.typed-array.float32-array.js";
import "core-js/modules/es.typed-array.float64-array.js";
import "core-js/modules/es.typed-array.int8-array.js";
import "core-js/modules/es.typed-array.int16-array.js";
import "core-js/modules/es.typed-array.int32-array.js";
import "core-js/modules/es.typed-array.uint8-array.js";
import "core-js/modules/es.typed-array.uint8-clamped-array.js";
import "core-js/modules/es.typed-array.uint16-array.js";
import "core-js/modules/es.typed-array.uint32-array.js";
import "core-js/modules/es.typed-array.at.js";
import "core-js/modules/es.typed-array.fill.js";
import "core-js/modules/es.typed-array.find-last.js";
import "core-js/modules/es.typed-array.find-last-index.js";
import "core-js/modules/es.typed-array.from.js";
import "core-js/modules/es.typed-array.of.js";
import "core-js/modules/es.typed-array.set.js";
import "core-js/modules/es.typed-array.sort.js";
import "core-js/modules/es.typed-array.to-locale-string.js";
import "core-js/modules/es.typed-array.to-reversed.js";
import "core-js/modules/es.typed-array.to-sorted.js";
import "core-js/modules/es.typed-array.with.js";
import "core-js/modules/es.weak-map.js";
import "core-js/modules/esnext.array.filter-out.js";
import "core-js/modules/esnext.array.filter-reject.js";
import "core-js/modules/esnext.array.group.js";
import "core-js/modules/esnext.array.group-by.js";
import "core-js/modules/esnext.array.group-by-to-map.js";
import "core-js/modules/esnext.array.group-to-map.js";
import "core-js/modules/esnext.array.is-template-object.js";
import "core-js/modules/esnext.array.last-index.js";
import "core-js/modules/esnext.array.last-item.js";
import "core-js/modules/esnext.array.unique-by.js";
import "core-js/modules/esnext.async-iterator.constructor.js";
import "core-js/modules/esnext.async-iterator.as-indexed-pairs.js";
import "core-js/modules/esnext.async-iterator.drop.js";
import "core-js/modules/esnext.async-iterator.every.js";
import "core-js/modules/esnext.async-iterator.filter.js";
import "core-js/modules/esnext.async-iterator.find.js";
import "core-js/modules/esnext.async-iterator.flat-map.js";
import "core-js/modules/esnext.async-iterator.for-each.js";
import "core-js/modules/esnext.async-iterator.from.js";
import "core-js/modules/esnext.async-iterator.indexed.js";
import "core-js/modules/esnext.async-iterator.map.js";
import "core-js/modules/esnext.async-iterator.reduce.js";
import "core-js/modules/esnext.async-iterator.some.js";
import "core-js/modules/esnext.async-iterator.take.js";
import "core-js/modules/esnext.async-iterator.to-array.js";
import "core-js/modules/esnext.bigint.range.js";
import "core-js/modules/esnext.composite-key.js";
import "core-js/modules/esnext.composite-symbol.js";
import "core-js/modules/esnext.data-view.get-uint8-clamped.js";
import "core-js/modules/esnext.data-view.set-uint8-clamped.js";
import "core-js/modules/esnext.function.demethodize.js";
import "core-js/modules/esnext.function.is-callable.js";
import "core-js/modules/esnext.function.is-constructor.js";
import "core-js/modules/esnext.function.metadata.js";
import "core-js/modules/esnext.function.un-this.js";
import "core-js/modules/esnext.iterator.as-indexed-pairs.js";
import "core-js/modules/esnext.iterator.chunks.js";
import "core-js/modules/esnext.iterator.concat.js";
import "core-js/modules/esnext.iterator.indexed.js";
import "core-js/modules/esnext.iterator.range.js";
import "core-js/modules/esnext.iterator.to-async.js";
import "core-js/modules/esnext.iterator.windows.js";
import "core-js/modules/esnext.iterator.zip.js";
import "core-js/modules/esnext.iterator.zip-keyed.js";
import "core-js/modules/esnext.json.is-raw-json.js";
import "core-js/modules/esnext.json.parse.js";
import "core-js/modules/esnext.json.raw-json.js";
import "core-js/modules/esnext.map.delete-all.js";
import "core-js/modules/esnext.map.emplace.js";
import "core-js/modules/esnext.map.every.js";
import "core-js/modules/esnext.map.filter.js";
import "core-js/modules/esnext.map.find.js";
import "core-js/modules/esnext.map.find-key.js";
import "core-js/modules/esnext.map.from.js";
import "core-js/modules/esnext.map.get-or-insert.js";
import "core-js/modules/esnext.map.get-or-insert-computed.js";
import "core-js/modules/esnext.map.includes.js";
import "core-js/modules/esnext.map.key-by.js";
import "core-js/modules/esnext.map.key-of.js";
import "core-js/modules/esnext.map.map-keys.js";
import "core-js/modules/esnext.map.map-values.js";
import "core-js/modules/esnext.map.merge.js";
import "core-js/modules/esnext.map.of.js";
import "core-js/modules/esnext.map.reduce.js";
import "core-js/modules/esnext.map.some.js";
import "core-js/modules/esnext.map.update.js";
import "core-js/modules/esnext.map.update-or-insert.js";
import "core-js/modules/esnext.map.upsert.js";
import "core-js/modules/esnext.math.clamp.js";
import "core-js/modules/esnext.math.deg-per-rad.js";
import "core-js/modules/esnext.math.degrees.js";
import "core-js/modules/esnext.math.fscale.js";
import "core-js/modules/esnext.math.iaddh.js";
import "core-js/modules/esnext.math.imulh.js";
import "core-js/modules/esnext.math.isubh.js";
import "core-js/modules/esnext.math.rad-per-deg.js";
import "core-js/modules/esnext.math.radians.js";
import "core-js/modules/esnext.math.scale.js";
import "core-js/modules/esnext.math.seeded-prng.js";
import "core-js/modules/esnext.math.signbit.js";
import "core-js/modules/esnext.math.sum-precise.js";
import "core-js/modules/esnext.math.umulh.js";
import "core-js/modules/esnext.number.clamp.js";
import "core-js/modules/esnext.number.from-string.js";
import "core-js/modules/esnext.number.range.js";
import "core-js/modules/esnext.object.iterate-entries.js";
import "core-js/modules/esnext.object.iterate-keys.js";
import "core-js/modules/esnext.object.iterate-values.js";
import "core-js/modules/esnext.observable.js";
import "core-js/modules/esnext.reflect.define-metadata.js";
import "core-js/modules/esnext.reflect.delete-metadata.js";
import "core-js/modules/esnext.reflect.get-metadata.js";
import "core-js/modules/esnext.reflect.get-metadata-keys.js";
import "core-js/modules/esnext.reflect.get-own-metadata.js";
import "core-js/modules/esnext.reflect.get-own-metadata-keys.js";
import "core-js/modules/esnext.reflect.has-metadata.js";
import "core-js/modules/esnext.reflect.has-own-metadata.js";
import "core-js/modules/esnext.reflect.metadata.js";
import "core-js/modules/esnext.set.add-all.js";
import "core-js/modules/esnext.set.delete-all.js";
import "core-js/modules/esnext.set.difference.js";
import "core-js/modules/esnext.set.every.js";
import "core-js/modules/esnext.set.filter.js";
import "core-js/modules/esnext.set.find.js";
import "core-js/modules/esnext.set.from.js";
import "core-js/modules/esnext.set.intersection.js";
import "core-js/modules/esnext.set.is-disjoint-from.js";
import "core-js/modules/esnext.set.is-subset-of.js";
import "core-js/modules/esnext.set.is-superset-of.js";
import "core-js/modules/esnext.set.join.js";
import "core-js/modules/esnext.set.map.js";
import "core-js/modules/esnext.set.of.js";
import "core-js/modules/esnext.set.reduce.js";
import "core-js/modules/esnext.set.some.js";
import "core-js/modules/esnext.set.symmetric-difference.js";
import "core-js/modules/esnext.set.union.js";
import "core-js/modules/esnext.string.at.js";
import "core-js/modules/esnext.string.cooked.js";
import "core-js/modules/esnext.string.code-points.js";
import "core-js/modules/esnext.string.dedent.js";
import "core-js/modules/esnext.symbol.custom-matcher.js";
import "core-js/modules/esnext.symbol.is-registered-symbol.js";
import "core-js/modules/esnext.symbol.is-registered.js";
import "core-js/modules/esnext.symbol.is-well-known-symbol.js";
import "core-js/modules/esnext.symbol.is-well-known.js";
import "core-js/modules/esnext.symbol.matcher.js";
import "core-js/modules/esnext.symbol.metadata.js";
import "core-js/modules/esnext.symbol.metadata-key.js";
import "core-js/modules/esnext.symbol.observable.js";
import "core-js/modules/esnext.symbol.pattern-match.js";
import "core-js/modules/esnext.symbol.replace-all.js";
import "core-js/modules/esnext.typed-array.from-async.js";
import "core-js/modules/esnext.typed-array.filter-out.js";
import "core-js/modules/esnext.typed-array.filter-reject.js";
import "core-js/modules/esnext.typed-array.group-by.js";
import "core-js/modules/esnext.typed-array.to-spliced.js";
import "core-js/modules/esnext.typed-array.unique-by.js";
import "core-js/modules/esnext.uint8-array.from-base64.js";
import "core-js/modules/esnext.uint8-array.from-hex.js";
import "core-js/modules/esnext.uint8-array.set-from-base64.js";
import "core-js/modules/esnext.uint8-array.set-from-hex.js";
import "core-js/modules/esnext.uint8-array.to-base64.js";
import "core-js/modules/esnext.uint8-array.to-hex.js";
import "core-js/modules/esnext.weak-map.delete-all.js";
import "core-js/modules/esnext.weak-map.from.js";
import "core-js/modules/esnext.weak-map.of.js";
import "core-js/modules/esnext.weak-map.emplace.js";
import "core-js/modules/esnext.weak-map.get-or-insert.js";
import "core-js/modules/esnext.weak-map.get-or-insert-computed.js";
import "core-js/modules/esnext.weak-map.upsert.js";
import "core-js/modules/esnext.weak-set.add-all.js";
import "core-js/modules/esnext.weak-set.delete-all.js";
import "core-js/modules/esnext.weak-set.from.js";
import "core-js/modules/esnext.weak-set.of.js";
import "core-js/modules/web.dom-collections.iterator.js";
import "core-js/modules/web.dom-exception.constructor.js";
import "core-js/modules/web.dom-exception.stack.js";
import "core-js/modules/web.dom-exception.to-string-tag.js";
import "core-js/modules/web.immediate.js";
import "core-js/modules/web.queue-microtask.js";
import "core-js/modules/web.self.js";
import "core-js/modules/web.structured-clone.js";
import "core-js/modules/web.url.js";
import "core-js/modules/web.url.can-parse.js";
import "core-js/modules/web.url.parse.js";
import "core-js/modules/web.url.to-json.js";
import "core-js/modules/web.url-search-params.js";
import "core-js/modules/web.url-search-params.delete.js";
import "core-js/modules/web.url-search-params.has.js";
import "core-js/modules/web.url-search-params.size.js";
class MyClass {
  constructor() {
    _defineProperty(this, "myProp", 42);
  }
}"
`;

exports[`buildPreset - Babel Transform should transform JSX in development 1`] = `
"import { jsxDEV as _jsxDEV } from "react/jsx-dev-runtime";
const element = /*#__PURE__*/_jsxDEV("div", {
  children: "Hello World"
}, void 0, false, {
  fileName: _jsxFileName,
  lineNumber: 1,
  columnNumber: 17
}, this);"
`;

exports[`buildPreset - Babel Transform should transform JSX in production 1`] = `
"import { jsx as _jsx } from "react/jsx-runtime";
const element = /*#__PURE__*/_jsx("div", {
  children: "Hello World"
});"
`;

exports[`buildPreset - Babel Transform should transform Object.hasOwn() with caller.library 1`] = `
"import _defineProperty from "@babel/runtime-corejs3/helpers/esm/defineProperty";
import _Object$hasOwn from "core-js-pure/stable/object/has-own.js";
class MyClass {
  constructor() {
    _defineProperty(this, "myProp", 42);
  }
}
console.log(_Object$hasOwn({
  a: 1
}, 'a') ? 'yes' : 'no');"
`;

exports[`buildPreset - Babel Transform should transform Object.hasOwn() with name==@babel/runtime 1`] = `
"import _defineProperty from "@babel/runtime-corejs3/helpers/esm/defineProperty";
import _Object$hasOwn from "core-js-pure/stable/object/has-own.js";
class MyClass {
  constructor() {
    _defineProperty(this, "myProp", 42);
  }
}
console.log(_Object$hasOwn({
  a: 1
}, 'a') ? 'yes' : 'no');"
`;

exports[`buildPreset - Babel Transform should transform Object.hasOwn() with name==rollup-plugin-babel 1`] = `
"import _defineProperty from "@babel/runtime-corejs3/helpers/esm/defineProperty";
import _Object$hasOwn from "core-js-pure/stable/object/has-own.js";
class MyClass {
  constructor() {
    _defineProperty(this, "myProp", 42);
  }
}
console.log(_Object$hasOwn({
  a: 1
}, 'a') ? 'yes' : 'no');"
`;

exports[`buildPreset - Babel Transform should transform Object.hasOwn() with usage-pure 1`] = `
"import _defineProperty from "@babel/runtime-corejs3/helpers/esm/defineProperty";
import _Object$hasOwn from "core-js-pure/stable/object/has-own.js";
class MyClass {
  constructor() {
    _defineProperty(this, "myProp", 42);
  }
}
console.log(_Object$hasOwn({
  a: 1
}, 'a') ? 'yes' : 'no');"
`;

exports[`buildPreset - Babel Transform should transform TypeScript to JavaScript 1`] = `"const x = 42;"`;

exports[`buildPreset - Babel Transform should transform class statics 1`] = `
"var _StaticEntity;
class StaticEntity extends Entity {}
_StaticEntity = StaticEntity;
StaticEntity.urlRoot = '/2/';
_StaticEntity.a = _StaticEntity.urlRoot;"
`;
